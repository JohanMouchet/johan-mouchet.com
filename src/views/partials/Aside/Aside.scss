@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.p-aside {
  flex-grow: 1;
  margin-top: t-vertical-rhythm(3);
  margin-bottom: t-vertical-rhythm(3);

  &--no-margin {
    margin-top: 0;
    margin-bottom: 0;
  }
}
