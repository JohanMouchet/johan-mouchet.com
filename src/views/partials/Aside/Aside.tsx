import React from "react";
import cx from "classnames";
import styles from "./Aside.module.scss";

type Props = {
  margin?: "no-margin";
  className?: string | string[] | { [key: string]: boolean };
  children: React.ReactNode;
};

const Aside: React.FC<Props> = ({ margin, className, children }) => {
  const classNames = cx("p-aside", margin && `p-main--${margin}`, className);

  return <aside className={classNames}>{children}</aside>;
};

export default Aside;
