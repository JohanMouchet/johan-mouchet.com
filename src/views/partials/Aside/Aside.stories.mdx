import { Meta, Preview, Story } from "@storybook/addon-docs/blocks";
import { select, text } from "@storybook/addon-knobs";
import Aside from ".";

<Meta title="Partials|Aside" component={Aside} />

# Aside

<Preview>
  <Story name="default">
    <div style={{ outline: "1px dotted" }}>
      <Aside
        margin={select("margin", {
          "": null,
          "no-margin": "no-margin",
        })}
      >
        {text("children", "children")}
      </Aside>
    </div>
  </Story>
</Preview>
