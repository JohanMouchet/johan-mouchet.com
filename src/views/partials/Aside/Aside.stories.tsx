import type { Meta, StoryObj } from "@storybook/react";
import { Aside } from "./Aside";

const meta = {
  title: "Partials/Aside",
  component: Aside,
} satisfies Meta<typeof Aside>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Example = {
  args: {
    children: "Aside",
  },
} satisfies Story;
