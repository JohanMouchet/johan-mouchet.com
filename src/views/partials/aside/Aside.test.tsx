import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { Aside } from "./Aside";

it("Renders <Aside /> without crashing", () => {
  expect(
    render(<Aside margin="no-margin">children</Aside>).container.firstChild
  ).toMatchSnapshot();
});
