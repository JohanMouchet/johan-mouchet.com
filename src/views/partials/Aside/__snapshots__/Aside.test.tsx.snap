// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Aside /> without crashing 1`] = `
<aside
  class="p-aside p-main--no-margin"
>
  children
</aside>
`;
